forum/users/serializers.py

```diff
@@ -0,0 +1,8 @@
+from rest_framework_simplejwt.serializers import TokenObtainPairSerializer
+
+class CustomTokenObtainPairSerializer(TokenObtainPairSerializer):
+    @classmethod
+    def get_token(cls, user, role='unassigned'):
+        token = super().get_token(user)
+        token['role'] = role
+        return token
```

Comment:
The `get_token` method should fetch the role from the user instance and add it to the token.
```
from rest_framework_simplejwt.serializers import TokenObtainPairSerializer

class CustomTokenObtainPairSerializer(TokenObtainPairSerializer):
    @classmethod
    def get_token(cls, user):
        # Get the token using the base class
        token = super().get_token(user)

        # Add custom claims (e.g., user role)
        token['role'] = getattr(user, 'role', 'unassigned')

        return token
```

Reply:
Done: reworked serializer completely according to new one-step auth concept